File: algorithms/CSharp/src/Sorts/selection-sort.cs
using System;

namespace Algorithms.Sorts
{
    public class SelectionSort
    {
        public static void Main()
        {
            int[] arr = { 800, 11, 50, 771, 649, 770, 240, 9 };
            Sort(arr);
            var result = string.Join(" ", arr);
            Console.WriteLine(result);
        }

        public static void Sort(int[] source)
        {
            int n = source.Length;

            // One by one move boundary of unsorted subarray 
            for (int i = 0; i < n - 1; i++)
            {
                // Find the minimum element in unsorted array 
                int min_idx = i;
                for (int j = i + 1; j < n; j++)
                {
                    if (source[j] < source[min_idx])
                    {
                        min_idx = j;
                    }
                }

                // Swap the found minimum element with the first 
                // element 
                int temp = source[min_idx];
                source[min_idx] = source[i];
                source[i] = temp;
            }
        }
    }
}

File: algorithms/JavaScript/sorting/selection-sort.js
function selectionSort(array) {

    // loop through all elements except last in array
    for(let i=0;i<array.length - 1;i++) {
        let min_index = i;

        // find minimum element index in unsorted array
        for(let j=i+1;j<array.length;j++) {
            if(array[j] < array[i]) {
                min_index = j;
            }

            // swap with element at minimum index
            let temp = array[min_index];
            array[min_index] = array[i];
            array[i] = temp;
        }
    }

    // return sorted array
    return array;
}

// test
console.log(selectionSort([9,8,7,6,5,4,3,2,1,0]))
